segmentstore/storage/impl/src/main/java/io/pravega/segmentstore/storage/impl/bookkeeper/BookKeeperLog.java

```diff
@@ -268,14 +268,11 @@ public QueueStats getQueueStatistics() {
     private void processWritesSync() {
         if (getWriteLedger().ledger.isClosed()) {
             // Current ledger is closed. Execute the rollover processor to safely create a new ledger. This will reinvoke
-            // the write processor upon finish, so
+            // the write processor upon finish, so the writes can be reattempted.
             this.rolloverProcessor.runAsync();
-        } else if (!processPendingWrites()) {
+        } else if (!processPendingWrites() && !this.closed.get()) {
```

Comment:
It's preemptively fixing another possible bug which hasn't bitten us yet. If you look at what `processPendingWrites` does, if it returns `false` then it already closed this object, thus this invocation would fail. We are just making sure it doesn't happen.